stripe/stripe-android
# [mc] Set pmo SFU instead of top-level SFU

⚠️ 🙅 Don't ship this until this API is GA
# Summary
If the checkbox is displayed, PaymentSheet sets the payment_method_options[card][setup_future_usage to "``off_session``" or "".
Removes the logic to disable save checkbox if non-reusable pm types are present

# Motivation
https://paper.dropbox.com/doc/Payment-Sheet-setup_future_usage-L0FGesFku5spzNZo7t5H6#:uid=645334800108472886154117&h2=Example

# Testing
Manual test plan:

1. Set sfu (happy path)
    1. Check “save” checkbox and pay with card 4242

2. Set sfu, then un-set SFU
    1. Check “save” checkbox and pay with card 4000 0000 0000 0002 (this test card always gets 
declined)
    2. Pay with card 4242 and un-check the “save” checkbox
    3. Payment should succeed, card should not be saved to the customer.

3. Set sfu, then use saved card
    1. Check “save” checkbox and pay with card 4000 0000 0000 0002 (this test card always gets declined)
    2. Pay with saved card.
    3. Payment should succeed, should not set up saved card for future use i.e. payment_method_options[card][setup_future_usage] should be ""

4. Set SFU, then don’t set SFU
    1. Check “save” checkbox and pay with card 4000 0000 0000 0002 (this test card always gets declined)
    2. Pay with iDEAL
    3. Payment should succeed, iDEAL (SEPA) should not be saved to the customer.

5. Set SFU with non-reusable PM types present

6. Don’t set SFU, then set SFU
    1. Un-check “save” checkbox and pay with card 4000 0000 0000 0002 (this test card always gets declined)
    2. Check the “save” checkbox and pay with card 4242


ios PR: https://github.com/stripe-ios/stripe-ios/pull/440
